engine/clients/redis/parser.py

```diff
@@ -0,0 +1,85 @@
+import json
+from collections import ChainMap
+from typing import Any, Dict, List, Optional, Text, Tuple
+
+from engine.base_client.parser import BaseConditionParser, FieldValue
+
+QueryParamsTuple = Tuple[Text, Dict[Text, Any]]
+
+
+class RedisConditionParser(BaseConditionParser):
+    def __init__(self) -> None:
+        super().__init__()
+        self.counter = 0
+
+    def build_condition(
+        self,
+        and_statements: List[QueryParamsTuple],
+        or_statements: List[QueryParamsTuple],
+    ) -> Tuple[Text, Dict[Text, Any]]:
+        if and_statements is None or 0 == len(and_statements):
+            and_clauses, and_params = [], []
+        else:
+            and_clauses, and_params = list(zip(*and_statements))
+        if or_statements is None or 0 == len(or_statements):
+            or_clauses, or_params = [], []
+        else:
+            or_clauses, or_params = list(zip(*or_statements))
+
+        clause = []
+        if and_clauses is not None and len(and_clauses) > 0:
```

Comment:
`and_clauses` check is redundant, according to the previous code it can be only list instance